File: api/src/main/java/com/dumbdogdiner/stickychat/api/messenger/MessageType.java
package com.dumbdogdiner.stickychat.api.messenger;

/**
 * Types of possible message packets.
 */
public enum MessageType {
    /**
     * Packet type for player messages in a channel.
     */
    SEND_MESSAGE,

    /**
     * Packet type for direct messages sent to players not found on the sender's server.
     */
    DIRECT_MESSAGE,

    /**
     * Packet type for players who have joined the network.
     */
    PLAYER_JOIN_NETWORK,

    /**
     * Packet type for players who have changed servers.
     */
    PLAYER_CHANGE_SERVER,

    /**
     * Packet type for channel creation.
     */
    CREATE_CHANNEL,

    /**
     * Packet type for channel deletion.
     */
    DELETE_CHANNEL
}

File: common/src/main/kotlin/com/dumbdogdiner/stickychatcommon/MessageType.kt
package com.dumbdogdiner.stickychatcommon

/**
 * Enum of possible plugin messages sent between server nodes, and to the Bungee plugin.
 */
enum class MessageType {
    MESSAGE,
    PRIVATE_MESSAGE,
    PRIVATE_MESSAGE_ACK,
    PRIVATE_MESSAGE_ERROR,
    MAIL
}
